src/oss-find-squats-lib/SquatChecker.cs

```diff
@@ -0,0 +1,119 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.CST.OpenSource.Model.Mutators;
+using Microsoft.CST.OpenSource.Shared;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.CST.OpenSource.FindSquats
+{
+    /// <summary>
+    /// The class to wrap around the mutators to get the dictionary of potential package mutations
+    /// instead of printing them to the console as is done in <see cref="FindSquatsTool"/>.
+    /// </summary>
+    public class SquatChecker
+    {
+        /// <summary>
+        /// The options for <see cref="SquatChecker"/>.
+        /// </summary>
+        public class Options
+        {
+            public int SleepDelay { get; set; }
+        }
+
+        public static NLog.ILogger Logger { get; set; } = NLog.LogManager.GetCurrentClassLogger();
+
+
+        /// <summary>
+        /// Generate the mutations for the package.
+        /// </summary>
+        /// <returns>A dictionary of each mutation, with a list of the reasons why this mutation showed up.</returns>
+        /// <exception cref="InvalidOperationException">Thrown if the package doesn't have a name.</exception>
+        public Dictionary<string, IList<string>> Mutate(PackageURL pUrl, BaseProjectManager manager)
```

Comment:
Does it make sense to move this functionality into the respective `BaseProjectManager` classes (or even the shared logic into the `BaseProjectManager` class itself)? That way the project managers are responsible for implementing (or reusing) the logic for mutating the package names and searching their public registry for the possible packages. That way you don't have to have project manager-specific code in this generic file (_i.e._, we don't have to check if the manager is NuGet to perform an additional check, like we're doing below)

Reply:
See my earlier comment on this. I'm uncomfortable putting the mutator functionality into the shared library because it is not shared by any other caller to the package managers other than find squats and changing shared implicitly changes all ~15 tools.

Further the mutators themselves may be used to mutate other things (for example we have a domain squat checker, and it's possible we could find other applications) so tightly coupling them to the package managers doesn't work in that aspect either.

My current thinking is to extend  the class based mutators as extension methods on BasePackageManager in the new find squats lib. 

For filtering, potentially the mutator itself could be responsible then for ensuring it's only run on the package managers that are applicable. Accomplishing the same goal of removing manager specific code from the squat checker itself. 

That said I don't totally disagree with keeping the manager specific complexity inside the squat checker as that is the only time it is relevant.